pkg/monitortests/clusterversionoperator/legacycvomonitortests/operators.go

```diff
@@ -10,21 +10,31 @@ import (
 	"k8s.io/client-go/kubernetes"
 
 	"github.com/openshift/origin/pkg/monitortests/clusterversionoperator/operatorstateanalyzer"
+	"github.com/sirupsen/logrus"
 
 	configv1 "github.com/openshift/api/config/v1"
+	clientconfigv1 "github.com/openshift/client-go/config/clientset/versioned"
 	"github.com/openshift/origin/pkg/monitor/monitorapi"
 	"github.com/openshift/origin/pkg/monitortestlibrary/platformidentification"
 	platformidentification2 "github.com/openshift/origin/pkg/monitortestlibrary/platformidentification"
 	"github.com/openshift/origin/pkg/test/ginkgo/junitapi"
+	exutil "github.com/openshift/origin/test/extended/util"
 	"k8s.io/client-go/rest"
 )
 
 // exceptionCallback consumes a suspicious condition and returns an
 // exception string if does not think the condition should be fatal.
-type exceptionCallback func(operator string, condition *configv1.ClusterOperatorStatusCondition, clientConfig *rest.Config) (string, error)
+type exceptionCallback func(operator string, condition *configv1.ClusterOperatorStatusCondition, eventInterval monitorapi.Interval, clientConfig *rest.Config) (string, error)
 
-func testStableSystemOperatorStateTransitions(events monitorapi.Intervals) []*junitapi.JUnitTestCase {
-	except := func(_ string, condition *configv1.ClusterOperatorStatusCondition, _ *rest.Config) (string, error) {
+func checkAuthenticationExceptions(condition *configv1.ClusterOperatorStatusCondition) bool {
+	if condition.Type == configv1.OperatorAvailable && condition.Status == configv1.ConditionFalse && (condition.Reason == "APIServices_Error" || condition.Reason == "APIServerDeployment_NoDeployment" || condition.Reason == "APIServerDeployment_NoPod" || condition.Reason == "APIServerDeployment_PreconditionNotFulfilled" || condition.Reason == "APIServices_PreconditionNotReady" || condition.Reason == "OAuthServerDeployment_NoDeployment" || condition.Reason == "OAuthServerRouteEndpointAccessibleController_EndpointUnavailable" || condition.Reason == "OAuthServerServiceEndpointAccessibleController_EndpointUnavailable" || condition.Reason == "WellKnown_NotReady") {
```

Comment:
I think this might be easier to read as a switch statement.

```go
if condition.Type == configv1.OperatorAvailable && condition.Status == configv1.ConditionFalse {
	switch condition.Reason {
	case "APIServices_Error", "APIServerDeployment_NoDeployment", "APIServerDeployment_NoPod",
		"APIServerDeployment_PreconditionNotFulfilled",  "APIServices_PreconditionNotReady",
		"OAuthServerDeployment_NoDeployment", "OAuthServerRouteEndpointAccessibleController_EndpointUnavailable", 
		"OAuthServerServiceEndpointAccessibleController_EndpointUnavailable", "WellKnown_NotReady":
		return true
	}
}
```

Reply:
I don't disagree.  I merely refactored this so I can reuse it and left it how it was.  I'm open to anyone making the change outside of this PR given how long this has been open and how much testing has gone into it.

Reply:
Sure thing, sounds good!